include/stl2/detail/algorithm/sort.hpp

```diff
@@ -24,81 +24,125 @@
-	safe_iterator_t<Rng> sort(Rng&& rng, Comp comp = {}, Proj proj = {})
-	{
-		return __stl2::sort(begin(rng), end(rng),
-			std::ref(comp), std::ref(proj));
-	}
+		template<RandomAccessIterator I, class Comp, class Proj>
+		requires Sortable<I, Comp, Proj>
+		static constexpr void final_insertion_sort(I first, I last, Comp &comp, Proj &proj)
+		{
+			if (distance(first, last) > introsort_threshold) {
+				detail::rsort::insertion_sort(first, first + introsort_threshold, comp, proj);
+				__sort_fn::unguarded_insertion_sort(first + introsort_threshold, last, comp, proj);
+			} else {
+				detail::rsort::insertion_sort(first, last, comp, proj);
+			}
+		}
 
-	namespace ext {
-#if STL2_WORKAROUND_GCC_79591
-		template<RandomAccessIterator I, Sentinel<I> S, class Comp = less,
-			class Proj = identity>
-		requires
-			Sortable<I, Comp, Proj>
-		I sort(I first, S sent, Comp comp = {}, Proj proj = {})
+		template <Integral I>
+		static constexpr auto log2(I n) {
+			STL2_EXPECT(n > 0);
+			I k = 0;
+			for (; n != 1; n /= 2) {
+				++k;
+			}
+			return k;
+		}
+	public:
+		template<RandomAccessIterator I, Sentinel<I> S, class Comp = less, class Proj = identity>
+		requires Sortable<I, Comp, Proj>
+		constexpr I operator()(I first, S sent, Comp comp = {}, Proj proj = {}) const
 		{
 			if (first == sent) {
 				return first;
 			}
-			auto last = __stl2::next(first, std::move(sent));
-			auto n = __stl2::distance(first, last);
-			detail::rsort::introsort_loop(first, last, detail::rsort::log2(n) * 2, comp, proj);
-			detail::rsort::final_insertion_sort(first, last, comp, proj);
+			auto last = next(first, std::move(sent));
+			auto n = distance(first, last);
+			__sort_fn::introsort_loop(first, last, __sort_fn::log2(n) * 2, comp, proj);
+			__sort_fn::final_insertion_sort(first, last, comp, proj);
 			return last;
 		}
 
-		template<RandomAccessRange Rng, class Comp = less, class Proj = identity>
-		requires
-			Sortable<iterator_t<Rng>, Comp, Proj>
-		safe_iterator_t<Rng> sort(Rng&& rng, Comp comp = {}, Proj proj = {})
+		template<RandomAccessRange R, class Comp = less, class Proj = identity>
+		requires Sortable<iterator_t<R>, Comp, Proj>
+		constexpr safe_iterator_t<R> operator()(R&& r, Comp comp = {}, Proj proj = {}) const
 		{
-			return __stl2::sort(begin(rng), end(rng),
-				std::ref(comp), std::ref(proj));
+			return (*this)(begin(r), end(r), std::ref(comp), std::ref(proj));
 		}
-#else  // STL2_WORKAROUND_GCC_79591
-		using __stl2::sort;
-#endif // STL2_WORKAROUND_GCC_79591
 
-		template<ForwardIterator I, Sentinel<I> S, class Comp = less,
-			class Proj = identity>
-		requires
-			Sortable<I, Comp, Proj>
-		I sort(I first, S last, Comp comp = {}, Proj proj = {})
+		/// Extension: sort using forward iterators
+		///
+		template<ForwardIterator I, Sentinel<I> S, class Comp = less, class Proj = identity>
+		requires Sortable<I, Comp, Proj>
+		constexpr I operator()(I first, S last, Comp comp = {}, Proj proj = {}) const
 		{
```

Comment:
[Not required] Would you mind merging these `Forward` overloads into the other overloads with `if constexpr`? Something like:
```c++
// Extension: supports forward foos
template<ForwardFoo x, /*... */>
requires /**/
/**/ operator()(/**/) const
{
  if constexpr (RandomAccessFoo<X>) {
    /**/
  } else {
    /**/
  }
}
```